content/en/tracing/custom_instrumentation/ruby.md

```diff
@@ -354,6 +354,128 @@ However, additional instrumentation provided by Datadog can be activated alongsi
+
+```
+gem 'opentelemetry-exporters-datadog'
+gem 'opentelemetry-api', '~> 0.5'
+gem 'opentelemetry-sdk', '~> 0.5'
+```
+
+- Or install the gems directly using:
+
+```
+gem install opentelemetry-api
+gem install opentelemetry-sdk
+gem install opentelemetry-exporters-datadog
+```
+
+### Usage
+
+Install the datadog processor and exporter in your application and configure the options. Then use the OpenTelemetry interfaces to produces traces and other information:
+
+```ruby
+require 'opentelemetry/sdk'
+require 'opentelemetry-exporters-datadog'
+
+# Configure the sdk with custom export
+OpenTelemetry::SDK.configure do |c|
+  c.add_span_processor(
+    OpenTelemetry::Exporters::Datadog::DatadogSpanProcessor.new(
+      OpenTelemetry::Exporters::Datadog::Exporter.new(
+        service_name: 'my_service', agent_url: 'http://localhost:8126'
+      )
+    )
+  )
+end
+
+# For propagation of datadog specific distibuted tracing headers,
+# set http propagation to the Composite Propagator
+OpenTelemetry::Exporters::Datadog::Propagator.auto_configure
+
+# To start a trace you need to get a Tracer from the TracerProvider
+tracer = OpenTelemetry.tracer_provider.tracer('my_app_or_gem', '0.1.0')
+
+# create a span
+tracer.in_span('foo') do |span|
+  # set an attribute
+  span.set_attribute('platform', 'osx')
+  # add an event
+  span.add_event(name: 'event in bar')
+  # create bar as child of foo
+  tracer.in_span('bar') do |child_span|
+    # inspect the span
+    pp child_span
+  end
+end
+```
+
+### Configuration Options - Datadog Agent URL
+
+By default the OpenTelemetry Datadog Exporter transmits traces to `http://localhost:8126`. You can configure the application to send traces to a diffent URL using the following environmennt variables:
+
+ - `DD_TRACE_AGENT_URL`: The `<host>:<port:` where you Datadog Agent is listening for traces. (e.g. `http://agent-host:8126`)
+
+These values can also be overridden at the trace exporter level:
+
+```ruby
+# Configure the sdk with custom export
+OpenTelemetry::SDK.configure do |c|
+  c.add_span_processor(
+    OpenTelemetry::Exporters::Datadog::DatadogSpanProcessor.new(
+      OpenTelemetry::Exporters::Datadog::Exporter.new(
+        service_name: 'my_service',
+        agent_url: 'http://dd-agent:8126',
+      )
+    )
+  )
+end
+```
+
+### Configuration Options - Tagging
+
+You can configure the application to automatically tag your Datadog exported traces, using the following environment variables:
+
+ - `DD_ENV`: Your application environment (e.g. `production`, `staging`, etc.)
+ - `DD_SERVICE`: Your application's default service name (e.g. `billing-api`)
+ - `DD_VERSION`: Your application version (e.g. `2.5`, `202003181415`, `1.3-alpha`, etc.)
+ - `DD_TAGS`: Custom tags in value pairs separated by `,` (e.g. `layer:api,team:intake`)
+    - If `DD_ENV`, `DD_SERVICE` or `DD_VERSION` are set, it will override any respective `env`/`service`/`version` tag defined in `DD_TAGS`.
```

Comment:
```suggestion
If `DD_ENV`, `DD_SERVICE`, or `DD_VERSION` is set, it will override any corresponding `env`, `service`, or `version` tag defined in `DD_TAGS`.
```

Reply:
👍 updated